src/Compilers/CSharp/Portable/Syntax/SyntaxKindFacts.cs

```diff
@@ -1048,9 +1050,15 @@ public static SyntaxKind GetPreprocessorKeywordKind(string text)
 
         public static IEnumerable<SyntaxKind> GetContextualKeywordKinds()
         {
-            for (int i = (int)SyntaxKind.YieldKeyword; i <= (int)SyntaxKind.WhenKeyword; i++)
+            ushort firstKeywordIdInclusive = (ushort)SyntaxKind.YieldKeyword;
+            ushort lastKeywordIdExclusive = (ushort)SyntaxKind.ElifKeyword;
+
+            for (ushort id = firstKeywordIdInclusive; id < lastKeywordIdExclusive; ++id)
             {
-                yield return (SyntaxKind)i;
+                if (Enum.IsDefined(typeof(SyntaxKind), id))
```

Comment:
Guess, you're the smart guy ;-) Otherwise enum values less then `ElifKeyword` are returned that aren't valid keyword id's.